pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -228,6 +228,26 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
 				continue
 			}
 
+			// Check whether container is present in state, there may be 3 reasons why it's not present:
+			// - container is in Be/Bu pod
+			// - kubelet just restarted - state has not been restored properly and is empty
+			// - container has been removed from state by RemoveContainer call (DeletionTimestamp is set)
+			if _, ok := m.state.GetCPUSet(containerID); !ok {
```

Comment:
Can we move this new block out to a separate PR and just make this PR about the persistent state?

Reply:
I would also suggest that we find a way to not do `AddContainer` in reconcile loop. May be it is enough if we do a reconcile on kubelet start/re-start if there is a cpu manager checkpoint file. If the checkpoint file is missing or if there are dangling Gu pods, only then we use `AddContainer`. The other option would be more drastic in that if the checkpoint file is missing, we drain the node? WDYT?

Reply:
@ConnorDoyle - ok
@balajismaniam 
> if we do a reconcile on kubelet start/re-start

I was thinking the same but then how many time should we run it ? `activePods` function after restart may not report anything for some time, and them might start reporting only some pods (already found by PLEG)

>If the checkpoint file is missing or if there are dangling Gu pods, only then we use AddContainer
CPUManager does not know (not should) know what pods/containers Policy is handling. It's Policy implementation that decides how and whom cpus should assign (Gu in case static Policy)